### PR TITLE
adds protocol tests for self closing tag and empty xml tag

### PR DESCRIPTION
**Pull request by skotambkar**
*Description of changes:*
* Adds protocol tests for self closing tag and empty xml tag

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/awsQuery/xml-lists.smithy [hunk 1]
```diff
@@ -31,7 +31,7 @@ operation XmlLists {
 apply XmlLists @httpResponseTests([
     {
         id: "QueryXmlLists",
-        documentation: "Serializes XML lists",
+        documentation: "Deserializes XML lists",
```
Comment (mtdowling) [inline smithy-aws-protocol-tests/model/awsQuery/xml-lists.smithy:34]: Note that these tests are not client-specific, so there's no need to change this language. If this is a concern, then I suggest we start using sentences like "Tests for XML list serialization".

File: smithy-aws-protocol-tests/model/awsQuery/xml-lists.smithy [hunk 2]
```diff
@@ -118,7 +118,29 @@ apply XmlLists @httpResponseTests([
                 }
             ]
         }
-    }
+    },
+    {
```
Comment (mtdowling) [inline smithy-aws-protocol-tests/model/awsQuery/xml-lists.smithy:122]: Lots of these tests seem to be testing for different possibilities that clients need to handle, but they don't seem like something a service could actually implement. Can you add a `client-only` tag to client-specific tests to indicate that they are client-specific tests? Like this one: https://github.com/awslabs/smithy/blob/master/smithy-aws-protocol-tests/model/restXml/http-query.smithy#L303

File: smithy-aws-protocol-tests/model/restXml/document-structs.smithy [hunk 3]
```diff
@@ -32,6 +32,7 @@ apply SimpleScalarProperties @httpRequestTests([
         body: """
               <SimpleScalarPropertiesInputOutput>
                   <stringValue>string</stringValue>
+                  <emptyStringValue></emptyStringValue>
```
Comment (JordonPhillips) [inline smithy-aws-protocol-tests/model/restXml/document-structs.smithy:35]: These also should probably be isolated with the client only tag

**Review by JordonPhillips — CHANGES_REQUESTED** [review]

File: smithy-aws-protocol-tests/model/restXml/document-structs.smithy (hunk 3 above)
Comment (skotambkar) [reply smithy-aws-protocol-tests/model/restXml/document-structs.smithy]: updated

**Review by JordonPhillips — APPROVED** [review]

**Pull request merged**